lib/internal/Magento/Framework/View/Element/AbstractBlock.php

```diff
@@ -665,6 +665,15 @@ public function toHtml()
         }
         $html = $this->_afterToHtml($html);
 
+        /** @var \Magento\Framework\DataObject */
+        $transportObject = new \Magento\Framework\DataObject(
```

Comment:
I looked closer at implementation. Please disregard my comment.

While it would be more correct to put triggering of both before- and after- events in layout, Magenot is not ready for that, since in many places Block::toHtml() is triggered directly.